### PR TITLE
✨ Handling quota limit error

### PR DESCRIPTION
**Pull request by MontaGhanmy**

### CODE REVIEW

File: tdrive/backend/node/src/services/documents/services/index.ts [hunk 1]
```diff
@@ -295,6 +301,8 @@ export class DocumentsService {
     context: DriveExecutionContext,
   ): Promise<DriveFile> => {
     try {
+      const userQuota = await this.userQuota(context);
```
Comment (shepilov) [inline tdrive/backend/node/src/services/documents/services/index.ts]: There is no need to calculate quota if it is disabled

File: tdrive/backend/node/src/services/documents/web/controllers/documents.ts [hunk 2]
```diff
@@ -80,7 +80,12 @@ export class DocumentsController {
         context,
       );
     } catch (error) {
+      console.log("🚀 ", JSON.stringify(error));
       logger.error({ error: `${error}` }, "Failed to create Drive item");
+      // if error code is 403, it means the user exceeded the quota limit
```
Comment (shepilov) [inline tdrive/backend/node/src/services/documents/web/controllers/documents.ts]: why do you need to catch it, and rethrow in controller?

File: tdrive/backend/node/src/services/documents/services/index.ts [hunk 3]
```diff
@@ -413,6 +432,7 @@ export class DocumentsService {
 
       return driveItem;
     } catch (error) {
+      console.log("🚀🚀 error:", error);
```
Comment (ericlinagora) [inline tdrive/backend/node/src/services/documents/services/index.ts]: 🥽

File: tdrive/backend/node/src/services/documents/services/index.ts [hunk 4]
```diff
@@ -773,6 +792,19 @@ export class DocumentsService {
       const driveItemVersion = getDefaultDriveItemVersion(version, context);
       const metadata = await getFileMetadata(driveItemVersion.file_metadata.external_id, context);
 
+      if (this.quotaEnabled) {
+        console.log("CREATE_VERSION::USER QUOTA ENABLED");
+        const userQuota = await this.userQuota(context);
+        const leftQuota = this.defaultQuota - userQuota;
+
+        if (metadata.size > leftQuota) {
+          console.log("CREATE_VERSION::NO_SPACE_LEFT");
```
Comment (ericlinagora) [inline tdrive/backend/node/src/services/documents/services/index.ts]: logs

**Review by shepilov — APPROVED** [review]

**Pull request merged**